doc/nrf/templates/sample_README.rst

```diff
@@ -61,34 +85,128 @@ Connect PIN1 to PIN2, then cut the connection again.
 
 User interface*
 ***************
+.. note::
+   Add button and LED assignments here, plus other information related to the user interface.
 
 Button 1:
-   Start or stop the disco show.
+   Does something.
 
-.. tip::
-   Add Button and LED assignments here, plus other information related to the user interface.
+Button 2:
+   Toggles something.
+
+LED 1:
+   On when connected.
+
+LED 2:
+   Indicates something.
+
+
+Configuration*
+**************
+
+.. note::
+   Always include this section if your sample can be configured by the user.
+   Start with the stock phrase that is included with ``|config|``.
+
+|config|
+
+Setup*
```

Comment:
I'm always confused about the Setup section, and I see some developers confused too. In nRF5 SDK, it would contain the path to source files, but this path could also be mentioned in Configuration. So do we really need to specify this Setup section?

Maybe Configuration could describe the required configuration steps (or none), with Configuration options describing the non-mandatory ones? Because I feel we have too many similar section here.

Reply:
We haven't had a setup section in NCS before, so I don't see a reason for anybody to get confused about this, especially since it clearly says that most samples won't need the section. In NCS, the path to the sample is mentioned in the Building and running section, because it isn't a part of the configuration.

The idea is to have a place for required setup steps that must be done before you can test the sample. This is NOT about setting configuration options (which is covered in the following sections), but an initial setup - for example, installing an external library (like here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/nrf9160/gps/README.html#using-the-sample-with-the-supl-client-library ) or setting up a server or account (like here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/nrf9160/http_application_update/README.html#setting-up-an-aws-s3-bucket ).

Tried to clarify the note.